card/mod/01_history/set/all/actions.rb

```diff
@@ -0,0 +1,47 @@
+# -*- encoding : utf-8 -*-
```

Comment:
I guess I had a picture that was much more oriented towards whitelisting
than blacklisting.

Let's say:

A depends on B,
F depends on G, and
Y depends on Z.

There are several ways we could create this whitelist.  It could be:

A,B,F,G,Y,and Z
A,F,Z
:mypackage (referring to some group of them)
...or whatever.

I don't know where exactly this configuration happens, but it's obviously
somewhere in the deck space.

Like with gems, "A depends on B" means that when you A, you must have B.
Unlike gems, when card mod A depends on card mod B, B needs to be loaded
first.

We could certainly get fancier than this, but it seems like this would be
very powerful already.

The reason I'm thinking of the whitelist and the dependency handling as
separate is that most of what I want to happen is this:

   1. mod whitelists are configured in deck
   2. at run time, mods are loaded (into ruby constant space) in standard
   order (based on mod paths config / alphabet)
   3. adjustments to mod inclusion order is worked out based on dependencies
   4. *all set modules are loaded immediately based on mod inclusion order
   5. other set modules are loaded dynamically based on mod inclusion order


In this model, they would seem to be different phases of the process.


On Mon, Mar 9, 2015 at 5:39 PM, Gerry Gleason <notifications@github.com>
wrote:

> In card/mod/01_history/set/all/actions.rb
> <https://github.com/wagn/wagn/pull/367#discussion_r26087613>:
>
> > @@ -0,0 +1,47 @@
> > +# -*- encoding : utf-8 -*-
>
> That's sort of what I was thinking with my request. Dependency handling is
> a lot more complex than just configuring a load order for the core mods,
> and that would be an opportunity to leave specific items out of the list.
> That might be enough for the standard modules. Maybe you could just have a
> single array that had all of them, core or deck looking first in the deck,
> then the gem.
>
> The full solution probably needs for the use cases to develop a bit so
> that we really need a more complex solution for dependencies.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wagn/wagn/pull/367/files#r26087613>.
>



-- 
Ethan McCutchen
One of the Wagneers, Wagn.org

Wagn. How pioneers roll.

s: ethan.mccutchen
t: @intogreater